ext/standaloneusers/xml/schema/CRM/Standaloneusers/User.xml

```diff
@@ -16,28 +16,81 @@
 
   <field>
     <name>id</name>
+    <title>UF Match ID</title>
     <type>int unsigned</type>
     <required>true</required>
     <comment>Unique User ID</comment>
     <html>
       <type>Number</type>
     </html>
+    <add>5.67</add>
   </field>
   <primaryKey>
     <name>id</name>
     <autoincrement>true</autoincrement>
   </primaryKey>
-
+  <field>
+    <name>domain_id</name>
+    <title>Domain ID</title>
+    <type>int unsigned</type>
+    <required>true</required>
+    <comment>Which Domain is this match entry for</comment>
+    <pseudoconstant>
+      <table>civicrm_domain</table>
+      <keyColumn>id</keyColumn>
+      <labelColumn>name</labelColumn>
+    </pseudoconstant>
+    <html>
+      <label>Domain</label>
+    </html>
+    <add>3.0</add>
+  </field>
+  <foreignKey>
+    <name>domain_id</name>
+    <table>civicrm_domain</table>
+    <key>id</key>
+    <add>3.0</add>
+  </foreignKey>
+  <field>
+    <name>uf_id</name>
+    <title>CMS ID</title>
+    <type>int unsigned</type>
+    <required>true</required>
+    <default>0</default>
+    <comment>UF ID. Redundant in Standalone. Needs to be identical to id.</comment>
```

Comment:
In my googling I found a pretty clever solution to this: instead of a real column, this could be a virtual generated column by adding this to the install.sql:

```
ALTER TABLE civicrm_uf_match ADD uf_id int unsigned
    GENERATED ALWAYS AS (id) VIRTUAL;
```
https://arctype.com/blog/mysql-generated-columns-guide/